discoverx/dx.py

```diff
@@ -52,13 +52,23 @@ def __init__(
         self.classification_threshold = self._validate_classification_threshold(classification_threshold)
         self.classification_table_name = classification_table_name
 
-        self.uc_enabled = self.spark.conf.get("spark.databricks.unityCatalog.enabled", "false")
+        self.uc_enabled = self.spark.conf.get("spark.databricks.unityCatalog.enabled", "false") == "true"
+        self.can_read_columns_table = self.can_read_columns_table()
 
         self.scanner: Optional[Scanner] = None
         self.classifier: Optional[Classifier] = None
 
         self.intro()
 
+    
```

Comment:
This is not necessary if we introduce save/load methods for the classification table. We should also make sure we use consistent naming so use `classification_table` instead of `columns_table`.